widgets/src/lib/InferenceWidget/shared/WidgetRecorder/WidgetRecorder.svelte

```diff
@@ -21,13 +22,31 @@
 	});
 
 	async function onClick() {
-		isRecording = !isRecording;
-		if (isRecording) {
-			recorder.start();
-			onRecordStart();
-		} else {
-			const blob = await recorder.stopRecording();
-			onRecordStop(blob);
+		try {
+			isRecording = !isRecording;
```

Comment:
Little nitpick: I would personally put this line just above the `try {`, as initial setting the recording state is not dependent on the success or failure of the operation down the line - but I guess that doesn't matter that much :D